src/widget/contexts.rs

```diff
@@ -146,40 +155,51 @@ impl<'a, 'b> LayoutCx<'a, 'b> {
         }
     }
 
-    pub fn add_event(&mut self, event: Event) {
+    pub fn add_event(&mut self, event: Message) {
```

Comment:
We should probably rename this and similar methods.

Reply:
Will do. However, looking at this now, I'm concerned I've introduced a layering violation; `Message` is a Xilem type, and that might make it harder to integrate other reactive layers. I'll file an issue on this.